### ISSUE TITLE
[V1.1] - GasLimit crash when provided more then 12 500 000

### ISSUE THREAD

**Issue by Roger-RumbleFish** [nervosnetwork/godwoken-web3#326]
On Godwoken v1.1 when we provide gasLimit higher then  12 500 000 the call instantly crash. Previously it try to make a call and after extending  12 500 000 it crash.

Subgraph have default value  50 000 000 
https://github.com/askucher/graph-node/blob/209d22a7f06d6bd42386047efbe3358013c7dd6a/chain/ethereum/src/runtime/runtime_adapter.rs#L27

![image](https://user-images.githubusercontent.com/82821191/167437418-fbb40af2-9e64-4472-98a1-5b83eaf5f03e.png)

Because of that each call on subgraph crash

also Geth default is 50 mln -> https://github.com/ethereum/go-ethereum/blob/e4b687cf462870538743b3218906940ae590e7fd/eth/ethconfig/config.go#L91

**Comment by RetricSu** [comment]
the max gas limit in polyjuice is  `12,500,000`, check the doc: https://github.com/nervosnetwork/godwoken-polyjuice/blob/main/docs/EVM-compatible.md#behavioral-differences-of-some-opcodes 

so when the subgraph `crash`, is there any error log?

**Comment by Roger-RumbleFish** [comment]
Here is the log 
```
graph-node_1      | May 09 14:07:51.514 INFO Possible reorg detected, retrying, error: failed to process trigger: block #3971 (0x2f85…e703), transaction d09781821d48f5d3ec48a5197526c485b0ed7cc17c14543e55dbc3e3010ce54e: Ethereum node returned an error when calling function "getLendingPool" of contract "LendingPoolAddressesProvider": RPC error: Error { code: InvalidParams, message: "invalid argument 0: eth_call -> callObj -> gas limit 0x2faf080 exceeds block gas limit of 12500000", data: None }
```

**Comment by Kuzirashi** [comment]
@Roger-RumbleFish 

Can you change the subgraph environment variable to limit gas?

https://github.com/graphprotocol/graph-node/pull/2333/files

eg. set `GRAPH_ETH_CALL_GAS` to `12500000`

**Comment by RetricSu** [comment]
> Hey with the issue, the problem is I'm not using 12.5 mln gas but a subgraph by default set it up to 50 mln. It shouldn't crash. Even geth got 50 as default. I know we cant use more but the problem is the default value for tools might set it up higher than that. People from the subgraph say that we shouldn't change this value. I'm not sure that it is even possible
>
> original post by @Roger-RumbleFish 

simply remove/enlarger the params validator for gasLimit is an easy fix but it conflicts with https://github.com/nervosnetwork/godwoken-web3/issues/259 

@Flouse any suggestion?

**Comment by Roger-RumbleFish** [comment]
I can't change it because they removed the parameter -> https://github.com/graphprotocol/graph-node/blob/2928450b0983ffcd74a5053cd3c3c6e1b2949fbb/NEWS.md#0250

**Comment by RetricSu** [comment]
> Even geth got 50 as default.

@Roger-RumbleFish can you post a reference for this geth's behavior?

**Comment by Roger-RumbleFish** [comment]
https://github.com/ethereum/go-ethereum/blob/e4b687cf462870538743b3218906940ae590e7fd/eth/ethconfig/config.go#L91

**Comment by RetricSu** [comment]
> > Hey with the issue, the problem is I'm not using 12.5 mln gas but a subgraph by default set it up to 50 mln. It shouldn't crash. Even geth got 50 as default. I know we cant use more but the problem is the default value for tools might set it up higher than that. People from the subgraph say that we shouldn't change this value. I'm not sure that it is even possible
> > original post by @Roger-RumbleFish
> 
> simply remove/enlarger the params validator for gasLimit is an easy fix but it conflicts with #259
> 
> subgraph tool	5000,0000
> polyjuice	1250,0000
> eth mainnet	3000,0000
> @Flouse any suggestion?

a simple thought: for max compatibility, we can set `5000,0000 ` as a rpc_gas_limit in the params validator, web3 only checks such limit, and let the polyjuice deal with the real gas limit `1250,0000`. then we descript such behavior in the docs, to distinguish the difference between the rpc param limit and the real block gas limit for developers. 

cc @Flouse @classicalliu @keroro520

**Comment by RetricSu** [comment]
testnet have been updated, feel free to open this issue if you still encounter trouble.
